nixos/doc/manual/release-notes/rl-2311.section.md

```diff
@@ -306,6 +306,21 @@ The module update takes care of the new config syntax and the data itself (user
   - `keepTerminfo` controls whether `TERMINFO` and `TERMINFO_DIRS` are preserved
     for `root` and the `wheel` group.
 
+- CoreDNS can now be built with external plugins by overriding `externalPlugins` and `vendorHash` arguments like this:
+```
+services.coredns = {
+  enable = true;
+  package = pkgs.coredns.override {
+    externalPlugins = [
+      {name = "fanout"; repo = "github.com/networkservicemesh/fanout"; version = "v1.9.1";}
+    ];
+    vendorHash = "<sha256 hash>";
+  };
+};
+```
+To get the necessary sha256 hash, set `vendorHash = "";`. The build will fail and produce your sha256 hash in the error message. You should copy that hash and use it for your `vendorHash`.
+Please note! If you use this feature, you will not receive updates to CoreDNS when you upgrade your system. To update CoreDNS, you'll have to update `vendorHash`; for that, set `vendorHash = "";` again and follow the steps above.
```

Comment:
Technically it's a bit more subtle than that. The newer version of `coredns` may attempt to build with the old `vendor`. It may even build successfully if nothing significant has changed. Or it may entirely fail to build. I'm not sure how much of that detail is worth putting in this note however. Just pointing it out.

Reply:
Yeah, this section was a little too detailed and repetitive IMO. I've streamlined it.